auth-backends/src/main/scala/bootstrap/rudder/plugin/AuthBackendsConf.scala

```diff
@@ -371,3 +379,193 @@ final class RudderOauth2Details(oauth2: OAuth2User, rudder: RudderUserDetail) ex
+  , authorizationRequestBaseUri : String
+) extends OAuth2AuthorizationRequestResolver {
+  val resolver = new DefaultOAuth2AuthorizationRequestResolver(clientRegistrationRepository, authorizationRequestBaseUri)
+
+  def cleanResolve(request: HttpServletRequest, registrationId: Option[String]) = {
+    val authzRequest =
+      try {
+        registrationId match {
+          case None     => resolver.resolve(request)
+          case Some(id) => resolver.resolve(request, id)
+        }
+      } catch {
+        case ex: Exception =>
+          val err = new OAuth2Error("authorization_request_build_error", ex.getMessage, null)
+          throw new OAuth2AuthenticationException(err, err.toString, ex)
+      }
+    if(null != authzRequest && AuthBackendsLogger.isDebugEnabled) {
+      AuthBackendsLogger.debug(s"Processing OAuth2/OIDC authorization to: ${authzRequest.getAuthorizationUri}")
+      if(AuthBackendsLogger.isTraceEnabled) {
+        val traceReq = s"\n  - authorization URI: ${authzRequest.getAuthorizationUri}" +
+                       s"\n  - grant type: ${authzRequest.getGrantType.getValue}" +
+                       s"\n  - response type: ${authzRequest.getResponseType.getValue}" +
+                       s"\n  - client ID: ${authzRequest.getClientId}" +
+                       s"\n  - redirect URI: ${authzRequest.getRedirectUri}" +
+                       s"\n  - scopes: ${authzRequest.getScopes.asScala.mkString(" ")}" +
+                       s"\n  - state: ${authzRequest.getState}" +
+                       s"\n  - additional parameters: ${authzRequest.getAdditionalParameters.asScala.toList.mkString("; ")}" +
+                       s"\n  - authorization request URI: ${authzRequest.getAuthorizationRequestUri}" +
+                       s"\n  - attributes: ${authzRequest.getAttributes.asScala.toList.mkString("; ")}"
+        AuthBackendsLogger.trace(s"  authorization request details: ${traceReq}")
+      }
+    }
+    authzRequest
+  }
+
+  override def resolve(request: HttpServletRequest): OAuth2AuthorizationRequest = {
+    cleanResolve(request, None)
+  }
+
+  override def resolve(request: HttpServletRequest, registrationId: String): OAuth2AuthorizationRequest = {
+    cleanResolve(request, Some(registrationId))
+  }
+
+}
+
+
+trait RudderUserServerMapping[R <: OAuth2UserRequest, U <: OAuth2User, T <: RudderUserDetail with U] {
+
+  def mapRudderUser(protocolName: String, delegateLoadUser: R => U, rudderUserDetailsService: RudderInMemoryUserDetailsService, userRequest: R, userBuilder: (U, RudderUserDetail) => T): T = {
+    val user = delegateLoadUser(userRequest)
+    val sub = user.getAttributes.get("sub").toString
+    AuthBackendsLogger.debug(s"Identifying ${protocolName} user info with sub: '${sub}' on rudder user base using login: '${user.getName}'")
+
+    // check that we know that user in our DB
+    val rudderUser = rudderUserDetailsService.loadUserByUsername(user.getName)
```

Comment:
so the heart of the problem was that here I was using `getPreferredUsername` which seems to be documented to be the attribute for the mapping, but no, it's `getName`